### PR TITLE
[Multi Hub] Tab User Profile Sample

### PR DESCRIPTION
**Pull request by Mohammed-MSFT**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |21| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
 |22| App Caching in Meetings | This sample shows how to improve the subsequent loading time of an App. |||[View][app-cache-meetings#nodejs]|
 |23| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
+|24| Tab User Profile In Outlook | Opens a profile card at a specified position to show profile information about a persona. ||[View][tab-user-profile#ts]|
```
Comment (ChetanSharma-msft) [inline README.md:57]: Opens a profile card at a specified position to show profile information about a persona. =>

This sample app shows how to open a user profile information card at a specified position in Outlook using personal tab app.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |21| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
```
Comment (ChetanSharma-msft) [inline README.md:54]: Build and deploy pipeline is missing for this sample and also ESBuild configuration is missing.

File: samples/tab-user-profile/ts/Manifest/manifest.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  "icons": {
+    "color": "color.png",
+    "outline": "outline.png"
+  },
+  "name": {
+    "short": "tab-user-profile",
+    "full": "tab-user-profile"
+  },
+  "description": {
+    "short": "Tab User Profile",
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/Manifest/manifest.json:22]: User profile in Outlook using personal tab app.

File: samples/tab-user-profile/ts/Manifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "full": "This sample shows how to open user profile card using personal tab in outlook."
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/Manifest/manifest.json:23]: This sample app shows how to open a user profile information card in Outlook using personal tab app.

File: samples/tab-user-profile/ts/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    ```
+ - Run your app
+
+    ```bash
+    npm start
+    ```
+3. Setup Manifest for Teams
+
+- **This step is specific to Teams.**
+    **Edit** the `manifest.json` contained in the `Manifest/` folder to replace with your Manifest-Id *everywhere* you see the place holder string `{{Manifest-Id}}`.
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/README.md:65]: How to get this {{Manifest-Id}} ? and what is the valid value for it?

File: samples/tab-user-profile/ts/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    **Edit** the `manifest.json` for `validDomains` and replace `{{domain-name}}` with base Url of your domain. E.g. if you are using ngrok it would be `https://1234.ngrok-free.app` then your domain-name will be `1234.ngrok-free.app`.
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/README.md:66]: This place holder does not exist in your manifest: {{domain-name}}`
Kindly add it.

File: samples/tab-user-profile/ts/assets/sample.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        "key": "TEAMS-SERVER-LANGUAGE",
+        "value": ""
+      },
+      {
+        "key": "TEAMS-SERVER-PLATFORM",
+        "value": ""
+      },
+      {
+        "key": "TEAMS-FEATURES",
+        "value": "tab"
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:31]: Add mail, outlook as well

File: samples/tab-user-profile/ts/assets/sample.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-tab-user-profile-ts",
+    "source": "officeDev",
+    "title": "Starter solution: Personal tab in TypeScript",
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:5]: User profile in Outlook using personal tab

File: samples/tab-user-profile/ts/assets/sample.json [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    "shortDescription": "Create React App personal tab in TypeScript",
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:6]: Get user profile information in Outlook using personal tab.

File: samples/tab-user-profile/ts/assets/sample.json [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/tab-user-profile/ts",
+    "longDescription": [
+      "Create React App personal tab in TypeScript"
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:9]: This sample app shows how to open a user profile information card at a specified position in Outlook using personal tab app.

File: samples/tab-user-profile/ts/assets/sample.json [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+      "Teams"
+    ],
+    "metadata": [
+      {
+        "key": "TEAMS-SAMPLE-SOURCE",
+        "value": "OfficeDev"
+      },
+      {
+        "key": "TEAMS-SERVER-LANGUAGE",
+        "value": ""
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:23]: Why this is empty, refer other sample

File: samples/tab-user-profile/ts/assets/sample.json [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+      },
+      {
+        "key": "TEAMS-SERVER-PLATFORM",
+        "value": ""
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:27]: Why this is empty, refer other sample

File: samples/tab-user-profile/ts/assets/sample.json [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        "key": "TEAMS-FEATURES",
+        "value": "tab"
+      }
+    ],
+    "thumbnails": [
+      {
+        "type": "image",
+        "order": 100,
+        "url": "https://raw.githubusercontent.com/OfficeDev/Microsoft-Teams-Samples/main/samples/tab-user-profile/ts/Images/meta-OS-profileGif.gif",
+        "alt": "Solution UX showing personal tab for user profile in outlook"
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/assets/sample.json:39]: Solution UX shows how to open a user profile information card in Outlook using personal tab app.

File: samples/tab-user-profile/ts/public/index.html [hunk 14]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="stylesheet" href="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-core/11.0.0/css/fabric.min.css" />
+    <title>Microsoft Teams Tab</title>
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/public/index.html:7]: This sample shows how to open a user profile information card at a specified position in Outlook using personal tab app.

File: samples/tab-user-profile/ts/src/components/App.tsx [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+// Licensed under the MIT License.
+
+import './App.css';
+import * as microsoftTeams from "@microsoft/teams-js";
+import { BrowserRouter,  Route, Routes } from "react-router-dom";
+import ProfileTab from "./profileTab";
+import Home from './home';
+
+/**
+ * The main app which handles the initialization and routing
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/App.tsx:11]: Single line comment.

File: samples/tab-user-profile/ts/src/components/home.css [hunk 16]
```diff
@@ -0,0 +1,346 @@
...
+    background-repeat: no-repeat;
+    background-size: 100%;
+    background-position: right;
+    background-color: #55A0E0;
+    width: 100%;
+    font-size: 44px;
+    height: 120px;
+    color: white;
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/home.css:10]: Check if you are using all of these CSS.

File: samples/tab-user-profile/ts/src/components/home.jsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
+// <copyright file="home.jsx" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// </copyright>
+
+import React from "react";
+import './home.css';
+
+// Mail Module in Teams Outlook default page home.tsx
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/home.jsx:9]: // User Profile Module in Teams Outlook default page home.tsx

File: samples/tab-user-profile/ts/src/components/profileTab.tsx [hunk 18]
```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import React, { useEffect, useState } from 'react';
+import { app, profile } from "@microsoft/teams-js";
+import validator from 'validator';
+import './Tab.css';
+
+// function for opening user profile card
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/profileTab.tsx:9]: // Function for opening user profile information card.

File: samples/tab-user-profile/ts/src/components/profileTab.tsx [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+function ProfileTab() {
+  const [selectedPeople, setSelectedPeople] = useState("");
+  const [requiredValidation, setRequiredValidation] = useState(false);
+
+  useEffect(() => {
+    app.initialize()
+  })
+
+  // set the input Email value to a state variable
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/profileTab.tsx:18]: // Set the input Email value to a state variable.

File: samples/tab-user-profile/ts/src/components/profileTab.tsx [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+  const setPeople = (e: any) => {
+    setSelectedPeople(e.target.value)
+  }
+
+  // Checks whether the platform supports Profile Module and shows the Profile card
```
Comment (ChetanSharma-msft) [inline samples/tab-user-profile/ts/src/components/profileTab.tsx:23]: // Checks whether the platform supports Profile Module and shows the user profile card.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

**Comment by ChetanSharma-msft** [comment]
@Mohammed-MSFT - Please run build pipeline.

File: samples/tab-user-profile/ts/README.md [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+
+**Open the app , now the app tab will appear. Enter any Email to get  a profile card at a specified position to show profile information about that persona**
+
+![MetaOsAppTab](Images/MetaOsAppTab.png)
+
+![MyProfileWindow](Images/MyProfileWindow.png)
+
+
+## Further Reading.
+[tab-user-profile](https://learn.microsoft.com/javascript/api/@microsoft/teams-js/profile?view=msteams-client-js-latest)
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/README.md]: Fixed

File: samples/tab-user-profile/ts/README.md [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+[TeamsJS capability support across Microsoft 365](https://learn.microsoft.com/microsoftteams/platform/m365-apps/teamsjs-support-m365#mail)
+[Extend Teams apps across Microsoft 365](https://learn.microsoft.com/microsoftteams/platform/m365-apps/overview)
+
+
+
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/README.md:100]: Fixed

File: samples/tab-user-profile/ts/Manifest/manifest.json [hunk 23]
```diff
@@ -0,0 +1,40 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/Manifest/manifest.json:2]: Fixed

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply README.md]: Fixed

Comment (Harikrishnan-MSFT) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (Harikrishnan-MSFT) [reply README.md:54]: Fixed

File: samples/tab-user-profile/ts/Manifest/manifest.json (hunk 3 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/Manifest/manifest.json]: Fixed

File: samples/tab-user-profile/ts/Manifest/manifest.json (hunk 4 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/Manifest/manifest.json]: Fixed

File: samples/tab-user-profile/ts/README.md (hunk 6 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/README.md:66]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 7 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 9 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/src/components/profileTab.tsx (hunk 20 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/profileTab.tsx]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 10 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 11 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 12 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/assets/sample.json (hunk 13 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/assets/sample.json]: Fixed

File: samples/tab-user-profile/ts/public/index.html (hunk 14 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/public/index.html]: Fixed

File: samples/tab-user-profile/ts/src/components/App.tsx (hunk 15 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/App.tsx]: Fixed

File: samples/tab-user-profile/ts/src/components/home.css (hunk 16 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/home.css]: Fixed

File: samples/tab-user-profile/ts/src/components/home.jsx (hunk 17 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/home.jsx]: Fixed

File: samples/tab-user-profile/ts/src/components/profileTab.tsx (hunk 18 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/profileTab.tsx]: Fixed

File: samples/tab-user-profile/ts/src/components/profileTab.tsx (hunk 19 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-user-profile/ts/src/components/profileTab.tsx]: Fixed

**Comment by Harikrishnan-MSFT** [comment]
Please run build pipeline - 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5356971545
image

File: README.md (hunk 2 above)
Comment (ChetanSharma-msft) [reply README.md:54]: We need to check and add deploy pipeline as well.

File: samples/tab-user-profile/ts/Manifest/manifest.json (hunk 23 above)
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/tab-user-profile/ts/Manifest/manifest.json:2]: @Harikrishnan-MSFT - What did we fix here?

**Review by ChetanSharma-msft — APPROVED** [review]
Added 2 comments.

File: samples/tab-user-profile/ts/Manifest/manifest.json (hunk 23 above)
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/tab-user-profile/ts/Manifest/manifest.json:2]: It's not working in PRE VERSION but working in normal outlook Desktop app.
image

**Comment by ChetanSharma-msft** [comment]
Duplicate of: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/852

**Pull request closed without merging**